conformance/utils/kubernetes/helpers.go

```diff
@@ -499,6 +530,40 @@ func GatewayListenersMustHaveConditions(t *testing.T, client client.Client, time
 	require.NoErrorf(t, waitErr, "error waiting for Gateway status to have conditions matching expectations on all listeners")
 }
 
+// GatewayListenerMustHaveConditions checks if the given listener of the specified gateway has all
+// the specified conditions.
+func GatewayListenerMustHaveConditions(t *testing.T, client client.Client, timeoutConfig config.TimeoutConfig, gwName types.NamespacedName, listenerName string, conditions []metav1.Condition) {
```

Comment:
I have the feeling this is very similar to `GatewayListenersMustHaveConditions`, maybe should use a different name here to disambiguate, or make one function part of the other